src/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj

```diff
@@ -46,3 +47,8 @@
 
   (stop [this context] context))
 
+(trapperkeeper/defservice jolokia-webservice
+  [[:WebserverService add-servlet-handler]]
+  (init [this context]
+    (add-servlet-handler (AgentServlet.) "/metrics/v2")
```

Comment:
I think we would at least need to use the webrouting service to get rid of the hardcoded `/metrics` here, e.g.:

https://github.com/puppetlabs/trapperkeeper-metrics/blob/master/src/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj#L44

You might still be able to slap the `/v2` on there while using the webrouting service's `get-route` fn to find the prefix.  If that works, it might be an acceptable solution.

The other option would be to see if we can't figure out how to write a bit of clojure code that wraps a servlet and makes it appear like a ring handler.  If we went that route we could pretty easily hard-code a whitelist of acceptable routes, which might be a good choice for ensuring that we don't expose the endpoints that are a security risk.  But it would be good to get some input from CS as to whether or not those endpoints would be useful enough in extreme circumstances to allow enabling them via config.